types/mydecimal_test.go

```
@@ -380,11 +380,16 @@ func (s *testMyDecimalSuite) TestFromString(c *C) {
 		{"1234500009876.5", "1234500009876.5", nil},
 		{"123E5", "12300000", nil},
 		{"123E-2", "1.23", nil},
+		{"1e18446744073709551615", "0", ErrBadNumber},
+		{"1e", "0", ErrBadNumber},
 	}
 	for _, ca := range tests {
 		var dec MyDecimal
 		err := dec.FromString([]byte(ca.input))
 		c.Check(err, Equals, ca.err)
+		if err != nil {
+			continue
```

Why do we ignore error here?